openedx/adg/admin_dashboard/performance/helpers.py

```diff
@@ -49,10 +53,30 @@ def get_course_grades_for_users(user_ids, course_id):
         user_id__in=user_ids, course_id=course_id
     ).only('user_id', 'percent_grade', 'passed_timestamp')
 
-    return {
+    grades_dict = {
         grade.user_id: {
             'grade_percentage': f'{convert_float_point_to_percentage(grade.percent_grade)}%',
             'passed_timestamp': grade.passed_timestamp
         }
         for grade in grades
     }
+
+    if len(user_ids) != len(grades):
+        course_grade_factory = CourseGradeFactory()
+
+        user_ids_without_grades = [user_id for user_id in user_ids if user_id not in grades_dict.keys()]
```

Comment:
`set(user_ids) - set(grades_dict.keys()` would be faster than iterating I guess.

Reply:
I have tested for 10k length and sets are taking .0021 and the above implementation is taking .0014.

Reply:
Yes, but iterating over the set would have been a little faster. 😛